core/src/main/java/org/apache/calcite/rel/rel2sql/SqlImplementor.java

```diff
@@ -1120,6 +1120,11 @@ public List<SqlNode> fieldList() {
     }
 
     void addOrderItem(List<SqlNode> orderByList, RelFieldCollation field) {
+      // If the field being ordered on is a Literal value, we can safely skip it.
```

Comment:
does this work if the field is an aliased literal, e.g. `date '2000-01-01' as d`? I'd add that as a scenario to the test.